roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -988,8 +988,61 @@ Int_t RooDataHist::getIndex(const RooAbsCollection& coord, bool fast) const {
   return calcTreeIndex(coord, fast);
 }
 
+std::string
+RooDataHist::declWeightArrayForCodeSquash(RooAbsArg const *klass, RooFit::Detail::CodeSquashContext &ctx) const
```

Comment:
Similar to my comment about the integrals, look at the implementation of `RooDataHist::weightFast()`: there is also a flag to not do the division by the bin widths, which is used in the RooHistFunc case.

This needs to be changed to support the RooHistFunc correctly.